tests/defences/trainer/test_adversarial_trainer_madry_pgd.py

```diff
@@ -0,0 +1,92 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+import numpy as np
+import pytest
+
+from art.defences.trainer.adversarial_trainer_madry_pgd import AdversarialTrainerMadryPGD
+from art.utils import load_mnist
+
+from tests.utils import master_seed
+
+logger = logging.getLogger(__name__)
+
+BATCH_SIZE = 10
+NB_TRAIN = 100
+NB_TEST = 100
+
+
+@pytest.fixture()
+def get_mnist_classifier(framework, image_dl_estimator):
+    def _get_classifier():
+        if framework in ["pytorch", "huggingface"]:
+            classifier, _ = image_dl_estimator()
+            master_seed(seed=1234, set_torch=True)
+        elif framework in ["tensorflow2", "tensorflow1"]:
+            classifier, _ = image_dl_estimator()
+            master_seed(seed=1234)
+        else:
+            return None
+
+        return classifier
+
+    return _get_classifier
```

Comment:
This should not be necessary, the fixture `image_dl_estimator` should be sufficient.
```suggestion
```

Reply:
agree, removed.